subsys/net/l2_wifi_if_conn/l2_wifi_conn.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(l2_wifi_mgr_conn, CONFIG_NET_CONNECTION_MANAGER_LOG_LEVEL);
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/wifi_mgmt.h>
+#include <zephyr/net/net_event.h>
+#include <net/wifi_mgmt_ext.h>
+#include <net/l2_wifi_connect.h>
+#include <zephyr/net/conn_mgr_connectivity.h>
+
+#define WIFI_SHELL_MGMT_EVENTS (NET_EVENT_WIFI_CONNECT_RESULT | \
+				NET_EVENT_WIFI_DISCONNECT_RESULT)
+
+static struct net_mgmt_event_callback net_l2_mgmt_cb;
+int connection_status;
+
+K_SEM_DEFINE(wait_for_conn_event_sem, 0, 1);
+
+static void net_l2_wifi_connect_result(struct net_mgmt_event_callback *cb)
+{
+	k_sem_give(&wait_for_conn_event_sem);
+}
+
+static void net_l2_wifi_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+				    uint32_t mgmt_event, struct net_if *iface)
+{
+	switch (mgmt_event) {
+	case NET_EVENT_WIFI_CONNECT_RESULT:
+		net_l2_wifi_connect_result(cb);
+		break;
+	case NET_EVENT_WIFI_DISCONNECT_RESULT:
+		break;
+	default:
+		break;
+	}
+}
+
+int net_l2_wifi_connect(struct conn_mgr_conn_binding *const binding)
+{
+	uint64_t time_stamp_start;
+	uint32_t ticks_elapsed;
+	int rc;
+	int sem_ret;
+
+	time_stamp_start = (uint64_t)k_uptime_ticks();
+	if (!binding) {
+		return -EINVAL;
+	}
+
+	while (true) {
+		rc = net_mgmt(NET_REQUEST_WIFI_CONNECT_STORED, binding->iface, NULL, 0);
```

Comment:
If connection succeeds here and I put the device in an RF chamber and it loses connectivity, what happens? Will it keep trying if persistence is enabled? And will the timeout apply during that retry? And will it give up immediately if persistence is disabled?

I'm not yet super familiar with the behavior of the `NET_REQUEST_WIFI_CONNECT_STORED` request, so I'm not quite sure which of the above desired behaviors will actually occur